### PR TITLE
Fix some rchk findings

### PR DESCRIPTION
**Pull request by romainfrancois**
See #4242 for details. original findings from rchk on dplyr 0.8.0.1: https://raw.githubusercontent.com/kalibera/cran-checks/master/rchk/results/dplyr.out

### CODE REVIEW

File: src/hybrid.cpp [hunk 1]
```diff
@@ -39,12 +39,14 @@ dplyr_hash_map<SEXP, hybrid_function>& get_hybrid_named_map() {
 }
 
 void hybrid_init(SEXP env, SEXP name, SEXP package, hybrid_id id) {
+  SEXP fun = PROTECT(Rf_findVarInFrame3(env, name, FALSE));
```
Comment (krlmlr) [inline src/hybrid.cpp]: Would `Shield<SEXP>` work here?

File: src/utils.cpp [hunk 2]
```diff
@@ -12,7 +12,8 @@
 using namespace Rcpp;
 
 SEXP child_env(SEXP parent) {
-  return Rf_eval(Rf_lang3(symbols::new_env, Rf_ScalarLogical(TRUE), parent), R_BaseEnv);
+  Shield<SEXP> call(Rf_lang3(symbols::new_env, Rf_ScalarLogical(TRUE), parent));
```
Comment (krlmlr) [inline src/utils.cpp]: Do we need to protect the scalar `TRUE` ?

Comment (romainfrancois) [reply src/utils.cpp]: Nope. Here is `ScalarLogical()` : 

```
INLINE_FUN SEXP ScalarLogical(int x)
{
    extern SEXP R_LogicalNAValue, R_TrueValue, R_FalseValue;
    if (x == NA_LOGICAL) return R_LogicalNAValue;
    else if (x != 0) return R_TrueValue;
    else return R_FalseValue;
}
```

File: src/hybrid.cpp (hunk 1 above)
Comment (romainfrancois) [reply src/hybrid.cpp]: Thanks.

File: inst/include/dplyr/hybrid/Expression.h [hunk 3]
```diff
@@ -228,25 +228,27 @@ class Expression {
...
+      val = PROTECT(rlang::quo_get_expr(val));
+      nprot++;
     }
 
     LOG_VERBOSE << "is_column_impl(false)";
+    bool result = false;
     if (is_column_impl(val, column, false)) {
-      return true;
+      result = true;
+    } else if (TYPEOF(val) == LANGSXP && Rf_length(val) == 1 && CAR(val) == symbols::desc && is_column_impl(CADR(val), column, true)) {
```
Comment (lionel-) [inline inst/include/dplyr/hybrid/Expression.h]: This branch is a bit long. Maybe extract in `is_desc()`?

**Comment by romainfrancois** [comment]
Closes #4242

**Pull request merged**